dds/DCPS/TypeObject.cpp

```diff
@@ -167,6 +169,37 @@ TypeIdentifier makeTypeIdentifier(const TypeObject& type_object)
   return TypeIdentifier();
 }
 
+
+void serialize_type_info(const TypeInformation& type_info, DDS::OctetSeq& seq)
+{
+  const DCPS::Encoding& encoding = XTypes::get_typeobject_encoding();
+  DCPS::Message_Block_Ptr data(new ACE_Message_Block(
+    DCPS::serialized_size(encoding, type_info)));
```

Comment:
This use of ACE_Message_Block is significantly different from the one below in `deserialize_type_info` and I think that one looks better to me.  Since the size is known and the data will have to go into the OctetSeq anyway, the first step (after determining the size) can be resizing the sequence.  Then the resized sequence's get_buffer returns a contiguous block of memory of the right size, and that can be wrapped by Message Block (and Data Block) without further heap allocation.

Since the pattern appears in multiple places (Message Block / OctetSeq conversions) maybe we need a common utility for it.  I don't think it should be directly built in to the Serializer, but we could have the Serializer use that same utility (for example, if Serializer had another constructor that took an OctetSeq).